tool2schema/parameter_schema.py

```diff
@@ -1,68 +1,57 @@
+from __future__ import annotations
+
 import re
-import typing
-from enum import Enum
-from inspect import Parameter, isclass
-from typing import Union
+from inspect import Parameter
+from typing import Optional, Union
 
 from tool2schema import Config
-
-TYPE_MAP = {
-    "int": "integer",
-    "float": "number",
-    "str": "string",
-    "bool": "boolean",
-    "list": "array",
-}
+from tool2schema.type_schema import EnumTypeSchema, TypeSchema
 
 
 class ParameterSchema:
     """
-    Automatically create a parameter schema given an instance of
-    inspect.Parameter and a function documentation string.
+    Automatically create a parameter schema given an instance of inspect.Parameter
+    and a function documentation string.
     """
 
-    def __init__(self, parameter: Parameter, index: int, config: Config, docstring: str = None):
+    def __init__(
+        self,
+        type_schema: TypeSchema,
+        parameter: Parameter,
+        index: int,
+        config: Config,
+        docstring: str = None,
+    ):
         """
         Create a new parameter schema.
 
+        :param type_schema: The type schema for the parameter (use `TypeSchema.create`)
         :param parameter: The parameter to create a schema for
         :param index: The index of the parameter in the function signature
+        :param config: Configuration settings to use when creating the schema
         :param docstring: The docstring for the function containing the parameter
         """
+        self.type_schema = type_schema
```

Comment:
For consistency

```python
self.type_schema: TypeSchema = type_schema
```

But I think you could probably delete the type hints for all of them here.

```python
self.type_schema = type_schema
self.parameter = parameter
self.index = index
self.config = config
self.docstring = docstring
```

Reply:
Should be done with 3354a38 (I removed all type hints)